roachtest: metamorphically enable decommissioning nudger

wenyihu6 opened the pull request:
Epic: none 
Release note: none

---
**roachtest: merge runDecommissionBench and runDecommissionBenchLong**

Previously, decommission roachtests with duration > 0 used
runDecommissionBenchLong, while the default used runDecommissionBench. It’s
unclear to me why both were needed since most of the logic was duplicated except
for minor differences. As a result, registering a decommission roachtest with a
longer duration and setting whileUpreplicating = true or slow writes would not
work, because the long version missed setup steps for additional nodes and slow
writes. It's unclear whether this was intentional. This commit merges the two
functions, keeping things more consistent.

---
**roachtest: metamorphically enable decommissioning nudger**

This commit enables the decommissioning nudger metamorphically for two
roachtests. I selected these two ad hoc — one is labeled as slow, and the other
has been intermittently failing on master to improve observability and help root
casue.

wenyihu6 committed:
roachtest: merge runDecommissionBench and runDecommissionBenchLong

Previously, decommission roachtests with duration > 0 used
runDecommissionBenchLong, while the default used runDecommissionBench. It’s
unclear to me why both were needed since most of the logic was duplicated except
for minor differences. As a result, registering a decommission roachtest with a
longer duration and setting whileUpreplicating = true or slow writes would not
work, because the long version missed setup steps for additional nodes and slow
writes. It's unclear whether this was intentional. This commit merges the two
functions, keeping things more consistent.

wenyihu6 committed:
roachtest: metamorphically enable decommissioning nudger

This commit enables the decommissioning nudger metamorphically for two
roachtests. I selected these two ad hoc — one is labeled as slow, and the other
has been intermittently failing on master to improve observability and help root
casue.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/154912)
<!-- Reviewable:end -->

wenyihu6 commented:
I also wanted to enable decommissioning nudger for https://github.com/cockroachdb/cockroach/issues/148884 and https://github.com/cockroachdb/cockroach/issues/151190. But these are decommission benchmarks, I'm not sure if we want to interfere with the benchmark results. I could add a separate roachtest for this, but I’m not sure it’s worth introducing another one just for this purpose.

wenyihu6 commented:
re: first commit, the diff between runDecommissionBenchLong and runDecommissionBench:  https://gist.github.com/wenyihu6/ece8ec7c02d37f821e79018aec4c3dd4/revisions#diff-ddff7c71c096781035d30b397776a83a3b6d46a3946e40f1fa6607fa91de7e07R121.

tbg reviewed (approved):
Thanks for getting this out so quickly! Could you run the tests affected by the first commit, for example via `./scripts/trigger-pr-roachtest.sh`?

For the two tests in which the setting is set, I've wondered if they run for long enough to see any effect of this. Maybe you could check (looking at recent nightlies should give enouhg of an idea)?

Conditionally approving.

> @tbg reviewed 1 of 1 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/154912#-:-OawUMIpELdn3BPxM_tR:b-msxe9w)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @wenyihu6)

___
*[`pkg/cmd/roachtest/tests/multi_store_remove.go` line 116 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/154912#-OawUJ_XBp52QMTillnq:-OawUJ_XBp52QMTillnr:b5f3sj6) ([raw file](https://github.com/cockroachdb/cockroach/blob/63cc60879ad3185232b7add0c6e8135040243f1e/pkg/cmd/roachtest/tests/multi_store_remove.go#L116)):*
> ```Go
> 
> 		if rng.Intn(2) == 0 {
> 			if _, err := conn.ExecContext(ctx, `SET CLUSTER SETTING kv.enqueue_in_replicate_queue_on_problem.interval = '10m'`); err != nil {
> ```

ditto about whether the test even runs long enough.
___
*[`pkg/cmd/roachtest/tests/decommission.go` line 1146 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/154912#-OawUFuMEwBjZBNLH1Tt:-OawUFuMEwBjZBNLH1Tu:b4l44av) ([raw file](https://github.com/cockroachdb/cockroach/blob/63cc60879ad3185232b7add0c6e8135040243f1e/pkg/cmd/roachtest/tests/decommission.go#L1146)):*
> ```Go
> 
> 			if rng.Intn(2) == 0 {
> 				run(db, `SET CLUSTER SETTING kv.enqueue_in_replicate_queue_on_problem.interval = '10m'`)
> ```

Does this test even run for ten minutes?


<!-- Sent from Reviewable.io -->

wenyihu6 commented:
Good point - decommission/slow and multi-store-remove are both pretty fast unless they fail to be decommissioned and fail the test - https://teamcity.cockroachdb.com/test/7943216852832997190?currentProjectId=Cockroach_Nightlies_Roachtests&expandTestHistoryChartSection=true. decommissionBench is usually the slow one, and drain-and-decommission is also pretty slow https://teamcity.cockroachdb.com/test/-4562091262767023949?currentProjectId=Cockroach_Nightlies_Roachtests&expandTestHistoryChartSection=true. I prefer to add the nudger to some of the decommissionBench tests to hopefully help root cause https://github.com/cockroachdb/cockroach/issues/148884 and https://github.com/cockroachdb/cockroach/issues/151190 as well, but I’m not sure how we feel about variability in the benchmark results. But I'm not sure if we actively look at the benchmark results.

wenyihu6 commented:
re: the first commit, I think it should only affect https://github.com/cockroachdb/cockroach/blob/1608511991185fd4d9bc967df3440e809c9f97a2/pkg/cmd/roachtest/tests/decommissionbench.go#L107 which is the only case with a non-zero duration, but should still run all the tests to be safe. Kicked off roachtest run on - https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/20576313?hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true. Thanks for the tips on the script command - didn't know we have that.

wenyihu6 committed:
roachtest: use decommission nudger for decommissionbench.go

This commit adds the decommissioning nudger as a new flavor in
`decommissionbench.go` and updates them to run weekly instead.

This commit also removes the metamorphically enabled decommissioning nudger from
`runDecommissionSlow`, since these roachtests typically complete in under 10
minutes and usually pass. The `decommissionBench`, on the other hand, runs
longer and is a better fit for this configuration.
